src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelControllerAdapter.java

```diff
@@ -4,16 +4,15 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawPanelControllerAdapter implements Job2dDriver {
 	private int startX = 0, startY = 0;
 
-	public MyAdapter() {
-		super();
-	}
+	private DrawPanelController drawerController = DrawerFeature.getDrawerController();
```

Comment:
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.